pkg/validation/spec/swagger.go

```diff
@@ -18,9 +18,13 @@ import (
 	"encoding/json"
 	"fmt"
 
+	jsonv2 "github.com/go-json-experiment/json"
 	"github.com/go-openapi/swag"
 )
 
+// Used by tests to selectively disable experimental JSON unmarshaler
+var UseExperimentalJSONUnmarshaling bool = true
```

Comment:
not sure if this is an intermediate state and gets cleaned up in a later commit, but prefer an unexported package var for this

Reply:
We can either use the public var in an internal package, or we can use two different unexported vars for v2 and v3. We can definitely switch to the latter if you think it's better, I personally don't see how it makes a difference.

Reply:
This comment must have been posted before I made the change to move this variable to `internal` package.

@liggitt is that okay? Would like it to live still be exposed so that the `spec3` tests can toggle it. (the `spec3` types depend on the openapi v2 types)